### PR TITLE
Add an example with local and remote resources as well as strategy-based tests

### PR DESCRIPTION
**Pull request by jluebbe**

### CODE REVIEW

File: labgrid/factory.py [hunk 1]
```diff
@@ -80,6 +80,23 @@ def _convert_to_named_list(self, data):
             assert 'cls' in item
         return result
 
+    def normalize_role(self, name, config):
+        role = config.get('role', name)
```
Comment (Emantor) [inline labgrid/factory.py]: This function is named `normalize_role` but does not do anything with the role. Rename to `normalize_config` and remove the role parameter?

File: labgrid/remote/client.py [hunk 2]
```diff
@@ -809,6 +823,21 @@ def create():
...
+def find_role_by_place(config, place):
+    for role, role_config in config.items():
+        resources, _ = target_factory.normalize_role(role, role_config)
+        remote_places = resources.get('RemotePlace', {})
+        remote_place = remote_places.get(place)
+        if remote_place:
+            return role
+
+def find_any_role_with_place(config):
+    for role, role_config in config.items():
```
Comment (Emantor) [inline labgrid/remote/client.py]: As above, iterating over the role is not necessary.

File: labgrid/remote/client.py [hunk 3]
```diff
@@ -809,6 +823,21 @@ def create():
     loop.run_until_complete(ready.wait())
     return session[0]
 
+def find_role_by_place(config, place):
+    for role, role_config in config.items():
```
Comment (Emantor) [inline labgrid/remote/client.py]: If we don't have to pass the role, this can be simplified as well.

File: labgrid/factory.py (hunk 1 above)
Comment (jluebbe) [reply labgrid/factory.py]: Renamed to normalize_config and dropped the name parameter.

File: labgrid/remote/client.py (hunk 2 above)
Comment (jluebbe) [reply labgrid/remote/client.py]: We still need the loop, but we don't need to pass the role name to normalize_config.

File: labgrid/remote/client.py (hunk 3 above)
Comment (jluebbe) [reply labgrid/remote/client.py]: The role name is part of the return value here as well.

**Pull request merged**